quipucords/tests/api/source/test_source.py

```diff
@@ -1710,3 +1710,1595 @@ def test_update_rhacs_green_path(self, django_client, rhacs_cred):
+            assert response_json == expected_response
+
+    def create_expect_201(self, data, django_client):
+        """Create a source, return the response as a dict."""
+        response = self.create(data, django_client)
+        assert response.status_code == status.HTTP_201_CREATED
+        return response.json()
+
+    @patch("api.source.view.start_scan", side_effect=dummy_start)
+    def create_with_query(self, query, data, django_client, start_scan):
+        """Create a source with query param, return the response as a dict."""
+        url = reverse("v2:source-list")
+        url += query
+        return django_client.post(url, json=data, headers=ACCEPT_JSON_HEADER)
+
+    def create_expect_201_with_query(self, query, data, django_client):
+        """Create a valid source with a query parameter."""
+        response = self.create_with_query(query, data, django_client)
+        assert response.status_code == status.HTTP_201_CREATED
+        return response.json()
+
+    def create_expect_400_with_query(
+        self, query, data, django_client, expected_response=None
+    ):
+        """Create an expect HTTP 400 with a query param."""
+        response = self.create_with_query(query, data, django_client)
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        if expected_response:
+            response_json = response.json()
+            assert response_json == expected_response
+
+    def get_source(self, django_client, params=None, url=None):
+        """Call the retrieve endpoint."""
+        if not url:
+            url = reverse("v2:source-list")
+        if params is not None:
+            return django_client.get(url, params=params)
+        else:
+            return django_client.get(url)
+
+    def update_source(self, django_client, data, source_id):
+        """Call the update endpoint."""
+        url = reverse("v2:source-detail", args=(source_id,))
+        return django_client.put(url=url, json=data)
+
+    def partially_update_source(self, django_client, data, source_id):
+        """Call the partially update endpoint."""
+        url = reverse("v2:source-detail", args=(source_id,))
+        return django_client.patch(url=url, json=data)
+
+    def source_object_factory(self, data, django_client, range_number):
+        """Create source objects and return list with name and ids."""
+        source_list = []
+        for index in range(range_number):
+            data["name"] = data["name"] + str(index)
+            response = self.create_expect_201(data, django_client)
+
+            source_info = {"name": data["name"], "id": response["id"]}
+            source_list.append(source_info)
+        return source_list
+
+    def test_validate_opts(self):
+        """Test the validate_opts function."""
+        source_type = DataSources.SATELLITE
+        options = {"use_paramiko": True}
+        with pytest.raises(ValidationError):
+            SourceSerializer.validate_opts(options, source_type)
```

Comment:
mmmm in this test case my confusion intensifies about the case for v1 vs v2. Isn't this `SourceSerializerV1`? 🤔

Reply:
if it is, shouldn't those tests live on the other `TestCase`, for serializer v1?